applications/conference/src/conf_config_req.erl

```diff
@@ -15,93 +15,142 @@
+    fetch_profile_config(JObj, requested_profile_name(JObj));
 handle_request(<<"Controls">>, JObj) ->
-    ConferenceName = kz_json:get_ne_value(<<"Profile">>, JObj, ?DEFAULT_PROFILE_NAME),
-    ControlsName = kz_json:get_ne_value(<<"Controls">>, JObj),
-    fetch_controls_config(JObj, ConferenceName, ControlsName).
+    %% NOTE: if this is not an internal page or default conference
+    %%  the requested_profile_name will only be the conference id.
+    %%  However, when the profile was returned by this application
+    %%  ecallmgr_fs_config 'fixed' set the controls profile names
+    %%  to {accountid}_{controlsname}.  Splitting requested_controls_name
+    %%  might give you the accountid and controls name internal key,
+    %%  and the profile name will be the conference id.
+    fetch_controls_config(JObj
+                         ,requested_profile_name(JObj)
+                         ,requested_controls_name(JObj)
+                         ).
 
 -spec fetch_profile_config(kz_json:object(), ne_binary()) -> 'ok'.
-fetch_profile_config(JObj, ?PAGE_PROFILE_NAME = ConfigName) ->
-    fetch_profile_config(JObj, 'undefined', ConfigName);
-fetch_profile_config(JObj, ConfigName) ->
-    [AccountId, ConferenceId] = binary:split(ConfigName, <<"_">>),
-    fetch_profile_config(JObj, AccountId, ConferenceId).
+fetch_profile_config(JObj, ProfileName) ->
+    case binary:split(ProfileName, <<"_">>) of
+        [AccountId, ConferenceId] -> fetch_profile_config(JObj, AccountId, ConferenceId);
+        _Else -> fetch_profile_config(JObj, 'undefined', ProfileName)
+    end.
 
 -spec fetch_profile_config(kz_json:object(), api_ne_binary(), ne_binary()) -> 'ok'.
-fetch_profile_config(JObj, AccountId, ?DEFAULT_PROFILE_NAME = ConfigName) ->
-    fetch_profile_config(JObj, AccountId, ConfigName, default_profile());
-fetch_profile_config(JObj, AccountId, ?PAGE_PROFILE_NAME = ConfigName) ->
-    fetch_profile_config(JObj, AccountId, ConfigName, page_profile());
+fetch_profile_config(JObj, _, ?PAGE_PROFILE_NAME = ProfileName) ->
+    fetch_profile_config(JObj, 'undefined', ProfileName, page_profile());
+fetch_profile_config(JObj, _, ?DEFAULT_PROFILE_NAME = ProfileName) ->
+    fetch_profile_config(JObj, 'undefined', ProfileName, default_profile());
+fetch_profile_config(JObj, 'undefined' = AccountId, ProfileName) ->
+    fetch_profile_config(JObj, AccountId, ProfileName, default_profile());
 fetch_profile_config(JObj, AccountId, ConferenceId) ->
+    lager:debug("looking up conference profile for ~s in account ~s"
+               ,[ConferenceId, AccountId]
+               ),
     Conference = get_conference(AccountId, ConferenceId),
     Profile = kapps_conference:profile(Conference),
-    Config = kapps_account_config:get_global(AccountId, ?CONFIG_CAT, [<<"profiles">>, Profile], default_profile()),
+    Config = kapps_account_config:get_global(AccountId, ?CONFIG_CAT, [<<"profiles">>, Profile]),
     fetch_profile_config(JObj, Conference, ConferenceId, Config).
```

Comment:
`Conference` is a `kapps_conference:conference()` here but in `fetch_profile_config/4` you changed it to be `api_ne_binary()`.